src/test/ui/rfc-2632-const-trait-impl/trait-where-clause.rs

```diff
@@ -0,0 +1,33 @@
+#![feature(const_fn_trait_bound)]
+#![feature(const_trait_impl)]
+
+trait Bar {}
+
+trait Foo {
+    fn a();
+    fn b() where Self: ~const Bar;
```

Comment:
A few more tests I think we should have:
* `where Self: ~const Foo`
* a const and a nonconst impl for such a trait
* a monomorphic use that actually runs it